compiler/ExtractBase.ml

```diff
@@ -486,6 +509,19 @@ let id_to_string (id : id) (ctx : extraction_ctx) : string =
         | Some lid -> ", loop: " ^ LoopId.to_string lid
       in
       "decreases clause for function: " ^ fun_name ^ loop
+  | TerminatesClauseId (fid, lid) ->
```

Comment:
I think we should change the names: I see F\* uses the name `decreases` while Lean uses the name `terminates`.
What annoys me is that depending on the backend `DecreasesClause` stands either for the decreasing measure (F\*) or for a tactic (Lean). I think we should use the name `DecreasesClause` and `DecreasesTac`, or `TerminatesClause` and `TerminatesTac`.

Reply:
I picked the names for consistency with the syntax, not the semantics:
- an F* termination *measure* (introduced by `decreases e`) is mapped via a DecreasesClauseId
- a Lean termination *measure* (introduced by `termination_by f x y z => e`) is mapped via a TerminatesClauseId
- a Lean termination *proof* (introduced by `decreasing_by <<tactic>>`) is mapped via a DecreasesClauseId 

Remember that in Lean4, the syntax is

```lean
termination_by f x y z => (x, y, z)
decreasing_by
  simp_wf
```

If you're arguing for a more semantic approach to this mapping, I would say we pick `TerminationMeasure` and `DecreasingProof` and map the former to `decreases` (F*) or `termination_by` (Lean), and the latter to `decreasing_by` (Lean only).

Thoughts?